app/Actions/Photo/Strategies/AddStandaloneStrategy.php

```diff
@@ -71,37 +73,100 @@ public function do(): Photo
-			$this->sourceFile->getOriginalExtension()
-		);
-		/** @var SizeVariantFactory $sizeVariantFactory */
-		$sizeVariantFactory = resolve(SizeVariantFactory::class);
-		$sizeVariantFactory->init($this->photo, $namingStrategy);
+			// Handle Google Motion Pictures
+			// We must extract the video stream from the original (local)
+			// file and stash it away, before the original file is moved into
+			// its (potentially remote) final position
+			if ($this->parameters->exifInfo->microVideoOffset !== 0) {
+				$tmpVideoFile = new TemporaryLocalFile(GoogleMotionPictureHandler::FINAL_VIDEO_FILE_EXTENSION, $this->sourceFile->getBasename());
+				$gmpHandler = new GoogleMotionPictureHandler();
+				$gmpHandler->load($this->sourceFile, $this->parameters->exifInfo->microVideoOffset);
+				$gmpHandler->saveVideoStream($tmpVideoFile);
+			} else {
+				$tmpVideoFile = null;
+			}
 
-		/**
-		 * Create size variant for original
-		 * Exception `IllegalOrderOfOperations` is never thrown, because we
-		 * have saved the photo above.
-		 *
-		 * @noinspection PhpUnhandledExceptionInspection
-		 */
-		$original = $sizeVariantFactory->createOriginal(
-			$this->parameters->exifInfo->width,
-			$this->parameters->exifInfo->height,
-			$this->sourceFile->getFilesize()
-		);
-		try {
-			$this->putSourceIntoFinalDestination($this->sourceFile, $original->short_path);
-		} catch (\Exception $e) {
+			// Create target file and symlink/copy/move source file to
+			// target.
+			// If import strategy request to delete the source file.
+			// `$this->sourceFile` will be deleted after this step.
+			// But `$this->sourceImage` remains in memory.
+			$targetFile = $this->namingStrategy->createFile(SizeVariant::ORIGINAL);
+			$streamStat = $this->putSourceIntoFinalDestination($targetFile);
+
+			// If we have a temporary video file from a Google Motion Picture,
+			// we must move the preliminary extracted video file next to the
+			// final target file
+			if ($tmpVideoFile !== null) {
+				$videoTargetPath =
+					pathinfo($targetFile->getRelativePath(), PATHINFO_DIRNAME) .
+					'/' .
+					pathinfo($targetFile->getRelativePath(), PATHINFO_FILENAME) .
+					$tmpVideoFile->getExtension();
+				$videoTargetFile = new FlysystemFile($targetFile->getDisk(), $videoTargetPath);
+				$videoTargetFile->write($tmpVideoFile->read());
+				$this->photo->live_photo_short_path = $videoTargetFile->getRelativePath();
+				$tmpVideoFile->close();
+				$tmpVideoFile->delete();
+				$tmpVideoFile = null;
+			}
+
+			// The original and final checksum may differ, if the photo has
+			// been rotated by `putSourceIntoFinalDestination` while being
+			// moved into final position.
+			$this->photo->checksum = $streamStat->checksum;
+			$this->photo->save();
+
+			// Create original size variant of photo
+			// If the image has been loaded (and potentially auto-rotated)
+			// take the dimension from the image.
+			// As a fallback (e.g. in case of videos) we use the EXIF data.
```

Comment:
Is that "in case of videos" in general or "in case of videos for which we did not extract a frame"?

Reply:
Your are right, the comment should be re-worded. It also holds for raw files which are acccepted but unsupported. What do think about this one?

```suggestion
			// As a fallback for media files from which no image could be extracted (e.g. unsupported file formats) we use the EXIF data.
```

Reply:
I think that's better, although I would probably add a concrete example (video files with no ffmpeg support but with exiftool support come to mind).